README.md

```diff
@@ -1,382 +1,144 @@
+Pipeline is launched automatically following each push in any branch:
+* Job CI: Checkout - Build - Test - Sonar - Upload artifact in github
+* Job CD: Checkout - Deploy in heroku
 
-  public ShoeCore get(BigInteger version){
-    return implementations.get(version);
-  }
+Consult workflow [pipeline in git repository](https://github.com/marouaneaba/core-abstract-sample/blob/next/.github/workflows/ci-cd.yml)
+Consult pipeline result [action](https://github.com/marouaneaba/core-abstract-sample/actions)
+## Testing
 
-  public void register(BigInteger version, ShoeCore implementation){
-    this.implementations.put(version, implementation);
-  }
+To launch your application's tests, run:
 
-}
-```
+    ./mvnw clean test
 
-</details>
+## Access endpoints to test
+The application exposes the API documentation, testable **/swagger-ui/** of the application :
 
-<details>
-<summary><code>ShoeCore</code></summary>
+http://[host:port]/swagger-ui/ (ex en local : http://localhost:8080/swagger-ui/)
 
-```java
-public interface ShoeCore {
+## APIs REST Endpoints
 
-  Shoes search(ShoeFilter filter);
+#### Fetch stock with state:
+````
+curl -X GET "http://localhost:8080/stock" -H "accept: application/json" -H "version: 3"
+````
+Response 
+200 body :
 
+```json
+{
+  "state": "SOME",
+  "shoes": [
+    {
+      "size": 1,
+      "color": "BLUE",
+      "quantity": 10
+    },
+    {
+      "size": 14,
+      "color": "BLUE",
+      "quantity": 0
+    }
+  ]
 }
 ```
-
-</details>
-
-<details>
-<summary><code>AbstractShoeCore</code></summary>
-
-```java
-public abstract class AbstractShoeCore implements ShoeCore {
-
-  @Autowired
-  private ShoeFacade shoeFacade;
-
-  @PostConstruct
-  void init(){
-
-    val version = Optional.ofNullable(this.getClass().getAnnotation(Implementation.class))
-                          .map(Implementation::version)
-                          .orElseThrow(() -> new FatalBeanException("AbstractShoeCore implementation should be annotated with @Implementation"));
-
-    shoeFacade.register(version, this);
-
-  }
-
+#### Replace stock with collection of shoes sent as input:
+````
+curl -X PATCH "http://localhost:8080/stock" -H "accept: */*" -H "version: 3" -H "Content-Type: application/json" -d 
+
+{
+    "shoes": [
+        {
+            "name": "basketsz",
+            "color": "BLUE",
+            "size": 1,
+            "quantity": 10
+        },
+        {
+            "name": "basketsz",
+            "color": "BLUE",
+            "size": 1,
+            "quantity": 0
+        }
+    ]
 }
-```
+````
 
-</details>
+Response: 
 
-<details>
-<summary><code>Implementation</code></summary>
 
-```java
-@Target({ElementType.TYPE})
-@Retention(RetentionPolicy.RUNTIME)
-@Component
-public @interface Implementation {
+* 204	No content
+* 400	Stock collection contains duplication shoes or stock capacity greater than 30 shoes
 
-  int version();
 
+#### Add shoes to the stock
+````
```

Comment:
Should have only three
Here's a TIP to ease the format : ``` BASH  resulting in : 
``` BASH
curl -X PATCH "http://localhost:8080/stock/shoes" -H "accept: application/json" -H "version: 3" -H "Content-Type: application/json" -d 
{ 
    "name": "basket",
    "color":"BLACK",
    "quantity":0,
    "size":0
}
```

Reply:
fix and is not push yet.